[ecobee] grant type refresh_token to URL https://api.ecobee.com/token failed with error code invalid_grant

I've been seeing the following error for about a week now and have been unable to resolve:

2020-10-10 11:54:19.459 [ERROR] [oauth2client.internal.OAuthConnector] - grant type refresh_token to URL https://api.ecobee.com/token failed with error code invalid_grant, description The authorization grant, token or credentials are invalid, expired, revoked, do not match the redirection URI used in the authorization request, or was issued to another client.
2020-10-10 11:54:19.464 [INFO ] [inding.ecobee.internal.api.EcobeeApi] - API: Exception getting access token: error='invalid_grant', description='The authorization grant, token or credentials are invalid, expired, revoked, do not match the redirection URI used in the authorization request, or was issued to another client.'

I've rebuilt the ecobee.com side and put the new apikey into the .things file.  I've restarted OH2 many times at this point.  Some posts suggested to delete "~/.java/.userPrefs", this also did not help.  I have not found a way to get OH to generate a new PIN request.

EDIT: I'm on OH 2.5.7 and running the new v2 ecobee binding.

--- a/bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/api/EcobeeApi.java
+++ b/bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/api/EcobeeApi.java
@@ -170,11 +170,22 @@ public class EcobeeApi implements AccessTokenRefreshListener {
             logger.info("API: The Ecobee authorization process threw an exception", e);
             ecobeeAuth.setState(EcobeeAuthState.NEED_PIN);
         } catch (OAuthResponseException e) {
-            logger.info("API: Exception getting access token: error='{}', description='{}'", e.getError(),
-                    e.getErrorDescription());
-            // How to handle the possible error codes?
+            handleOAuthException(e);
         }
         return isAuthorized;
+    }
+
+    private void handleOAuthException(OAuthResponseException e) {
+        if ("invalid_grant".equalsIgnoreCase(e.getError())) {
+            // Usually indicates that the refresh token is no longer valid and will require reauthorization
+            logger.warn("API: Received 'invalid_grant' error response. Please reauthorize application with Ecobee");
+            deleteOAuthClientService();
+            createOAuthClientService();
+        } else {
+            // Other errors may not require reauthorization and/or may not apply
+            logger.warn("API: Exception getting access token: error='{}', description='{}'", e.getError(),
+                    e.getErrorDescription());
+        }
     }
 
     @Override

